File: Data Structures & Algorithms/lowest-common-ancestor-in-binary-search-tree/submission-2.js
class Solution {
	/**
	 * @param {TreeNode} root
	 * @param {TreeNode} p
	 * @param {TreeNode} q
	 * @return {TreeNode}
	 */
	lowestCommonAncestor(root, p, q) {
		if (!root) return null;
		if (p.val < root.val && q.val < root.val)
			return this.lowestCommonAncestor(root.left, p, q);
		if (p.val > root.val && q.val > root.val)
			return this.lowestCommonAncestor(root.right, p, q);
		return root;
	}
}

File: Data Structures & Algorithms/valid-binary-search-tree/submission-2.py

class Solution:
    def isValidBST(self, root: Optional[TreeNode]) -> bool:

        def validate(left: float, node: Optional[TreeNode], right: float):
            if not node:
                return True
            if not (left < node.val < right):
                return False
            return validate(left, node.left, node.val) and validate(
                node.val, node.right, right
            )

        return validate(float("-inf"), root, float("inf"))
